### PR TITLE
Substrait support for ParquetExec round trip for simple select

### PR DESCRIPTION
**Pull request by xinlifoobar**
## Which issue does this PR close?

Parts of #9347. To close this ticket, at least the translations between substrait types to arrow types translations and statistics support should be added.

## Are there any user-facing changes?

Yes, but substrait for physical plans is not functional yet.

### CODE REVIEW

File: datafusion/substrait/tests/testdata/data.parquet
Comment (xinlifoobar) [inline datafusion/substrait/tests/testdata/data.parquet:1]: Those are generated from `data.csv` under the same folder using `pandas`, e.g.,

```
dataset = pd.read_csv("datafusion/substrait/tests/testdata/data.csv")
dataset.to_parquet("datafusion/substrait/tests/testdata/data.parquet")
```

File: datafusion/substrait/src/physical_plan/producer.rs [hunk 1]
```diff
@@ -55,15 +60,56 @@ pub fn to_substrait_rel(
...
+                    names,
+                    r#struct: Some(type_info),
+                }),
                 filter: None,
                 best_effort_filter: None,
                 projection: Some(MaskExpression {
-                    select: None,
-                    maintain_singular_struct: false,
+                    select: select_struct,
+                    // FIXME: duckdb set this to true, but it's not clear why.
```
Comment (alamb) [inline datafusion/substrait/src/physical_plan/producer.rs:110]: > 		// fixme: whatever this means

😆

File: datafusion/substrait/tests/testdata/data.parquet
Comment (alamb) [reply datafusion/substrait/tests/testdata/data.parquet:1]: Can you please put a README in that directory with this information? It is important context but would likely be hard to find after this PR

Perhaps https://github.com/apache/datafusion/tree/main/datafusion/substrait/tests/testdata/README.md

**Review by alamb — APPROVED** [review]
Thank you @xinlifoobar  -- this looks great to me!

File: datafusion/substrait/tests/testdata/data.parquet
Comment (xinlifoobar) [reply datafusion/substrait/tests/testdata/data.parquet:1]: Done

File: datafusion/substrait/tests/testdata/Readme.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Apache DataFusion Substrait Test Data
```
Comment (alamb) [inline datafusion/substrait/tests/testdata/Readme.md:20]: ❤️

**Review by alamb — APPROVED** [review]
Thanks again @xinlifoobar

**Pull request merged**